docs/migration-guide.md

```diff
@@ -0,0 +1,332 @@
+const elem = await fixture(html`
+  <some-elem>
+    <some-other-elem></some-other-elem>
+  </some-elem>
+`);
+```
+
+However, waiting for all nested `updateComplete`s means `fixture()` can be a bit slower. Therefore, tests that were relying on waiting for an event fired during the initial render will likely fail due to the event now firing during the `fixture()` call.
+
+Those tests will need to be reworked to not rely on events fired during `render()`.
+
+### Step 4: Use Browser Interaction Helpers
+
+For tests which were manually calling `elem.click()` or `elem.hover()`, use the `clickElem()` and `hoverElem()` helpers. These helpers will trigger a real browser interaction using the mouse.
+
+Before:
+
+```javascript
+it('should focus', async() => {
+  const elem = await fixture(html`<button>hello</button>`);
+  elem.click();
+});
+```
+
+After:
+
+```javascript
+import { clickElem } from '@brightspace-ui/testing';
+
+it('should focus', async() => {
+  const elem = await fixture(html`<button>hello</button>`);
+  await clickElem(elem);
+});
+```
+
+If tests were manually focusing and/or dispatching keyboard events, use the `focusElem()` and `sendKeysElem()` helpers.
+
+> **Note:** `sendKeysElem()` will focus on the element before sending the keys, so calling `focusElem()` beforehand isn't necessary.
+
+```javascript
+it('should press ENTER', async() => {
+  const elem = await fixture(html`<input type="text">`);
+  elem.focus();
+  const eventObj = document.createEvent('Events');
+  eventObj.initEvent('keypress', true, true);
+  eventObj.keyCode = '13';
+  elem.dispatchEvent(eventObj);
+});
+```
+
+After:
+
+```javascript
+import { focusElem, sendKeysElem } from '@brightspace-ui/testing';
+
+it('should focus', async() => {
+  const elem = await fixture(html`<button>hello</button>`);
+  await sendKeysElem(elem, 'press', 'Enter');
+});
+```
+
+### Step 5: Use `d2l-test-runner` Binary
+
+Switch to `d2l-test-runner` from `web-test-runner`/`wtr` binaries in any `package.json` scripts.
+
+By default, `d2l-test-runner` will:
+
+* Run in Chrome, Firefox and Safari (except `vdiff` just runs in Chrome)
+* Enable the `--node-resolve` flag
+* Look for tests in `./test/**/*.test.js` (using the `test` group)
+* Suppress `ResizeObserver` errors (a common reason repos use a config file)
+
+Due to these new defaults, if the repo is using a `web-test-runner.config.js`, it may no longer be necessary and could be removed. If the config file is necessary, rename it to `d2l-test-runner.config.js`.
+
+Before
+
+```yml
+  "scripts": {
+    "test:unit": "web-test-runner --files \"./test/**/*.test.js\" --node-resolve"
+  }
+```
+
+After:
+
+```yml
+  "scripts": {
+    "test:unit": "d2l-test-runner"
+  }
+```
+
+## vdiff Tests
+
+Breaking changes were made between the `@brightspace-ui/visual-diff` library to `@brightspace-ui/testing`'s vdiff functionality. These changes will require a migration, some of which can be done automatically but unfortunately some is manual.
```

Comment:
Not tied to my suggestion. Just something about what you have now initially seemed off, though when I reread it it seems fine.

```suggestion
Breaking changes were made between the `@brightspace-ui/visual-diff` library and `@brightspace-ui/testing`'s vdiff functionality. These changes will require a migration, which is unfortunately only partially automated.
```

Reply:
I like yours better!